### ISSUE TITLE
return NotImplemented

### ISSUE THREAD

**Issue by ydawei** [pytorch/pytorch#7266]
This is minor and not urgent, but in batch normalization: https://github.com/pytorch/pytorch/blob/76d3c30783d3e808b070f8350d9f102bb2396944/torch/nn/modules/batchnorm.py#L48

It seems to be `raise NotImplementedError`, according to the [python doc](https://docs.python.org/3.7/library/constants.html#NotImplemented).

**Comment by soumith** [comment]
@ydawei one never uses the _BatchNorm base class. So we never ran into this :), we should correct it.
